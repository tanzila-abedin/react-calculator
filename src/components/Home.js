import '../style/Home.css';

const Home = () => (
  <div className="home container p-5">
    <p className="text-center welcome-text" data-testid="welcomeText">
      Welcome to Math Magician website
    </p>
  </div>
);

export default Home;
