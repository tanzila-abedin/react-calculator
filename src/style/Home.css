.welcome-text {
  font-weight: 700;
  text-align: center;
  font-size: 40px;
  font-family: Hack, sans-serif;
  letter-spacing: 5px;
}
