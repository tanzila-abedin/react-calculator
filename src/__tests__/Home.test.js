import { expect, test } from '@jest/globals';
import { render } from '@testing-library/react';
import renderer from 'react-test-renderer';
import { StaticRouter } from 'react-router-dom';
import Home from '../components/Home';

test('Snapshot of Home Component', () => {
  const tree = renderer
    .create(
      <StaticRouter>
        <Home />
      </StaticRouter>,
    )
    .toJSON();
  expect(tree).toMatchSnapshot();
});

test('Displays welcome text on homepage', async () => {
  const home = render(
    <StaticRouter>
      <Home />
    </StaticRouter>,
  );

  const welcomeText = await home.findByTestId('welcomeText');
  expect(welcomeText.textContent).toBe('Welcome to Math Magician website');
});

test('For presence of classnames in p', async () => {
  const home = render(
    <StaticRouter>
      <Home />
    </StaticRouter>,
  );

  const welcomeText = await home.findByTestId('welcomeText');
  expect(welcomeText.className).toContain('text-center', 'welcome-text');
});
