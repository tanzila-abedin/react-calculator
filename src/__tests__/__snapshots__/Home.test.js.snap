// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Snapshot of Home Component 1`] = `
<div
  className="home container p-5"
>
  <p
    className="text-center welcome-text"
    data-testid="welcomeText"
  >
    Welcome to Math Magician website
  </p>
</div>
`;
